src/libs/vehicle/ardupilot/ardupilot.ts

```diff
@@ -386,6 +390,60 @@ export abstract class ArduPilotVehicle<Modes> extends Vehicle.AbstractVehicle<Mo
     return true
   }
 
+  /**
+   * Helper function for commanding takeoff
+   * @param {number} altitude
+   */
+  async _takeoff(altitude: number): Promise<void> {
+    this.sendCommandLong(MavCmd.MAV_CMD_NAV_TAKEOFF, 0, 0, 0, 0, 0, 0, altitude)
+
+    // check command is accepted
+    let timeoutReachedCount = false
```

Comment:
I agree.
This will be needed in all functions that send commands.

What we can do is to have a callback registering for listening to ACK messages with it's specific ID, so that each function that wants to know when the ACK message of it's specific command ID arrives can register a callback to it and then react accordingly.

Let me know if you want to discuss that approach more in depth. This is probably the most important addition, as it will impact all other commands.

Reply:
I would suggest to remove the ACK part from this PR to avoiding dragging it longer. And deal with such task/feature in a different PR. For now, if sending command long does what it needs, we should let the validation part for latter. Since ACK is out of the scope of this PR and can get really crazy.